packages/core/src/browser/shell/tab-bars.ts

```diff
@@ -614,10 +616,12 @@ export class TabBarRenderer extends TabBar.Renderer {
                 this.selectionService.selection = NavigatableWidget.is(widget) ? { uri: widget.getResourceUri() } : widget;
             }
 
+            const contextKeyServiceOverly = this.contextKeyService?.createOverlay([['isTerminalTab', widget && '_terminalId' in widget]]);
```

Comment:
I would rather use _terminalId_ property from the interface rather than the one from the implementation. 
That is usually what we do with the is() methods. I do not see better way to identify if the widget is a terminal widget.